lib/features/wallet_screen/add_contact_page/add_contact_page.dart

```diff
@@ -0,0 +1,145 @@
+import 'package:threedpass/core/polkawallet/bloc/app_service_cubit.dart';
+import 'package:threedpass/core/utils/show_text_snackbar.dart';
+import 'package:threedpass/core/widgets/buttons/card_elevated_button.dart';
+import 'package:threedpass/core/widgets/d3p_scaffold.dart';
+import 'package:threedpass/core/widgets/input/textformfield/textformfield.dart';
+import 'package:threedpass/core/widgets/paddings.dart';
+import 'package:threedpass/features/wallet_screen/add_contact_page/bloc/contacts_bloc.dart';
+import 'package:threedpass/features/wallet_screen/add_contact_page/domain/entities/contact.dart';
+
+@RoutePage()
+class AddContactPage extends StatelessWidget {
+  AddContactPage({final Key? key}) : super(key: key);
+
+  final _formKey = GlobalKey<FormState>();
+  late final AppService appService;
+  final TextEditingController nameTextController = TextEditingController();
+  final TextEditingController addressTextController = TextEditingController();
+
+  @override
+  Widget build(final BuildContext context) {
+    appService = BlocProvider.of<AppServiceLoaderCubit>(context).state;
+    return D3pScaffold(
+      appbarTitle: 'add_contact_appbar_title',
+      body: SafeArea(
+        child: Padding(
+          padding: const EdgeInsets.all(16),
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.center,
+            children: [
+              Form(
+                key: _formKey,
+                child: Row(
+                  children: [
+                    Container(
+                      width: 100,
+                      child: D3pTextFormField(
+                        labelText: 'contact_name'.tr(),
+                        hintText: 'contact_name_hint'.tr(),
+                        controller: nameTextController,
+                        validator: _validateInputName,
+                      ),
+                    ),
+                    const SizedBox(width: 16),
+                    Expanded(
+                      child: D3pTextFormField(
+                        labelText: 'contact_address'.tr(),
+                        hintText: 'contact_address_hint'.tr(),
+                        controller: addressTextController,
+                        validator: _validateInputAddress,
+                      ),
+                    ),
+                  ],
+                ),
+              ),
+              const SizedBoxH16(),
+              D3pCardElevatedButton(
+                text: 'add_contact'.tr(),
+                onPressed: () => onTapAddContact(context),
+                iconData: Icons.add,
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  Future<void> onTapAddContact(final BuildContext context) async {
+    final bool isCorrect =
+        await checkAddressAndNotify(addressTextController.text);
+    if (isCorrect) {
+      // Check user input
+      if (_formKey.currentState != null && _formKey.currentState!.validate()) {
```

Comment:
I think form validation should be placed before address check, because it's sync and has higher priority